src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -397,250 +337,352 @@ export default function parseManifest(
-              adaptation.attributes.height;
+          if (
+            type === "text" &&
+            adaptationChildren.accessibility &&
+            isHardOfHearing(adaptationChildren.accessibility)
+          ) {
+            closedCaption = true;
           }
 
-          if (representation.attributes.maxPlayoutRate != null) {
-            parsedRepresentation.maxPlayoutRate =
-              representation.attributes.maxPlayoutRate;
-          } else if (adaptation.attributes.maxPlayoutRate != null) {
-            parsedRepresentation.maxPlayoutRate =
-              adaptation.attributes.maxPlayoutRate;
+          if (
+            type === "audio" &&
+            adaptationChildren.accessibility &&
+            isVisuallyImpaired(adaptationChildren.accessibility)
+          ) {
+            audioDescription = true;
           }
 
-          if (representation.attributes.maximumSAPPeriod != null) {
-            parsedRepresentation.maximumSAPPeriod =
-              representation.attributes.maximumSAPPeriod;
-          } else if (adaptation.attributes.maximumSAPPeriod != null) {
-            parsedRepresentation.maximumSAPPeriod =
-              adaptation.attributes.maximumSAPPeriod;
+          let adaptationID : string;
+          if (adaptation.attributes.id != null) {
+            adaptationID = adaptation.attributes.id;
+          } else {
+            let idString = type;
+            if (adaptation.attributes.language) {
+              idString += `-${adaptation.attributes.language}`;
+            }
+            if (closedCaption) {
+              idString += "-cc";
+            }
+            if (audioDescription) {
+              idString += "-ad";
+            }
+            if (adaptation.attributes.contentType) {
+              idString += `-${adaptation.attributes.contentType}`;
+            }
+            if (adaptation.attributes.codecs) {
+              idString += `-${adaptation.attributes.codecs}`;
+            }
+            if (adaptation.attributes.mimeType) {
+              idString += `-${adaptation.attributes.mimeType}`;
+            }
+            if (adaptation.attributes.frameRate) {
+              idString += `-${adaptation.attributes.frameRate}`;
+            }
+
+            if (idString.length === type.length) {
+              idString += representations.length ?
+                ("-" + representations[0].id) : "-empty";
+            }
+
+            adaptationID = "adaptation-" + idString;
           }
 
-          if (representation.attributes.mimeType != null) {
-            parsedRepresentation.mimeType =
-              representation.attributes.mimeType;
-          } else if (adaptation.attributes.mimeType != null) {
-            parsedRepresentation.mimeType =
-              adaptation.attributes.mimeType;
-          }
+          const parsedAdaptationSet : IParsedAdaptation = {
+            id: adaptationID,
+            representations,
+            type,
+            role: adaptationRole,
```

Comment:
Wut

Reply:
The aim is to keep in track which adaptation is 'main' adaptation, only during parsing, to merge other main adaptations.